---
title: 'CDSBMexico: remember to apply for BioC2019 travel scholarships'
author: L. Collado-Torres
date: '2019-02-28'
slug: cdsbmexico-remember-to-apply-for-bioc2019-travel-scholarships
categories:
  - rstats
  - Conference
  - UNAM
tags:
  - Diversity
  - Bioconductor
header:
  caption: ''
  image: ''
  preview: yes
---



<p><em>This blog post was first published at the <a href="https://comunidadbioinfo.github.io/post/cdsbmexico-remember-to-apply-for-bioc2019-travel-scholarships/">CDSBMexico website</a>.</em></p>
{{% tweet "1101339389395943424" %}}
<p>About 10 months ago <a href="http://lcolladotor.github.io/2018/04/19/latin-american-r-bioconductor-developers-workshop-2018/#.XHiaE1NKg0o">we announced our plans</a> to start a new community of R/Bioconductor developers in Mexico and Latin America. The National Bioinformatics Node (<a href="http://www.nnb.unam.mx/">NNB-UNAM</a> in Spanish) <a href="http://congresos.nnb.unam.mx/">has organized workshops</a> in bioinformatics and R since 2006 and more frequently since 2010. Last year, 2018, there were <a href="http://congresos.nnb.unam.mx/TIB2018/">five simultaneous one week workshops</a>:</p>
<ul>
<li>Introduction to R and RStudio</li>
<li>Exploratory data analysis (EDA) of biological data with R</li>
<li>EDA of RNA-seq data and differential expression studies</li>
<li>Genome assembly and annotation from high-throughput sequencing data</li>
<li>And our <a href="http://congresos.nnb.unam.mx/TIB2018/r-bioconductor-developers-workshop-2018/"><em>Latin American R/Bioconductor Developers Workshop</em></a></li>
</ul>
<p>Overall it was a success as described in <a href="https://comunidadbioinfo.github.io/post/a-recap-of-cdsb-2018-the-start-of-a-community/"><em>A recap of CDSB 2018: the start of a community</em></a>. However, we are not done yet! ğŸ’ªğŸ½As we have stated several times, our goal is to turn R users into developers and to increase the representation of Mexicans and Latin Americans at other R events.</p>
<div id="bioc2019" class="section level3">
<h3>BioC2019</h3>
<p><img src="http://bioconductor.org/images/logo_bioconductor.gif" /></p>
<p>The one R event that we followed as a guide last year is the Bioconductor yearly meeting, or <em>BioC</em>. This year, <a href="https://bioc2019.bioconductor.org/index">BioC2019</a> will be held during June 24 to 27 at New York City in the United States. If you are an academic the registration fee is 300 USD prior to May 24th. As of today (Feb 28th), a round trip from Mexico City on June 23 to 28th is about 320-410 USD<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. The <a href="https://bioc2019.bioconductor.org/travel-accommodations">accommodation options listed</a> start at 118 USD plus tax a night. As you can see, these numbers add up fairly fast which is a limiting factor for scientists from Mexico (and Latin American in general) to attend <em>BioC2019</em>.</p>
</div>
<div id="travel-scholarships" class="section level3">
<h3>Travel scholarships</h3>
<p>However, we want to remind you that there are travel scholarships available! And we, <a href="https://comunidadbioinfo.github.io/"><em>CDSB</em></a>, want to help you! ğŸ™ŒğŸ½</p>
<p>All the details about the travel scholarship and different types of submission proposals are available on the <a href="https://bioc2019.bioconductor.org/call-for-abstracts">BioC2019</a> website. But let us summarize the process for you.</p>
<ol style="list-style-type: decimal">
<li>You have to apply for either a <strong>talk</strong>, <strong>poster</strong> or <strong>workshop</strong>.</li>
<li>You need a professional website, GitHub profile or your CV to be available online.</li>
<li>You need to answer two questions:</li>
</ol>
<ul>
<li>How have you participated in the Bioconductor project in the past?</li>
<li>What do you hope to get out of the conference?</li>
</ul>
<p>For a talk or poster proposal you have a maximum of 300 words while workshop proposals involve filling out a template.</p>
<p>This is quite a bit of work, but the payoff is immense! You will compete for the opportunity to travel basically for free: free registration and hotel, plus $500 for travel expenses. This can be a career changing opportunity as it was <a href="https://speakerdeck.com/lcolladotor/cdsbmexico">the case for L. Collado-Torres (see his keynote slides)</a> and others.</p>
</div>
<div id="get-started-on-your-scholarship-application" class="section level3">
<h3>Get started on your scholarship application!</h3>
<p>We know that many of us have a tendency to do things at the last minute. However, we hope ğŸ™ğŸ½
that you start your application materials as soon as you can. Remember that the deadline is <strong>March 15th</strong> and <strong>no exceptions</strong> are made.</p>
<div id="website" class="section level4">
<h4>Website</h4>
<p>If you donâ€™t have a professional website, we recommend that you make one following <a href="https://www.emilyzabor.com/tutorials/rmarkdown_websites_tutorial.html">Emily Zaborâ€™s tutorial</a>. This is the tutorial that one of your fellow CDSBMexico 2018 attendees used to create his website: <a href="https://joseaia.github.io/">joseaia.github.io</a>. Alternatively, check out the <a href="https://leanpub.com/universities/courses/jhu/cbds-intro-r">Chromebook Data Science: intro to R course (available for free if you canâ€™t afford it)</a> which includes a lesson that walks you through this process.</p>
</div>
<div id="bioconductor-project-participation" class="section level4">
<h4>Bioconductor project participation</h4>
<p>Next, we know that each of you worked at least in one R package during CDSBMexico 2018 such as <a href="https://comunidadbioinfo.github.io/post/r-gene-regulatory-interaction-formulator-for-inquiring-networks/">rGriffin (check the blog post)</a>. Maybe you even submitted your package to Bioconductor. At least, you met and interacted with longtime Bioconductor members such as Martin Morgan and Benilton Carvalho. Since then, you might have worked on other R packages, shiny applications, etc. You might have also answered questions on the Bioconductor support website, reported bugs or opened issues in existing packages, joined the developers mailing list, joined the Bioconductor Slack workspace, etc. Ultimately, you could potentially present a poster or give a talk about the package you worked on last summer, but youâ€™ll need to coordinate with your team members and/or lab principal investigator.</p>
</div>
<div id="benefitting-from-bioc2019" class="section level4">
<h4>Benefitting from BioC2019</h4>
<p>As for what you hope to get out of the conference, well, we will remind you that BioC has different workshops that can help you advance your R skills and knowledge. Furthermore, thereâ€™ll be scientific talks from excellent bioinformatics researchers. Then, once you return from the conference, you can spread the knowledge you acquired in a variety of ways: implementing it on your work, teaching at your local university or research institute, participating in the Bioconductor community (Slack, the support website, contributing an R package, etc), presenting at a local R meetup, and participating in other forums. You can also benefit personally from attending the conference by networking with other users and developers that will attend it. From it, you might start new collaborations, find about graduate school programs, find potential postdoc advisors, among other opportunities. Hey, you might help us out with future CDSBMexico events!</p>
</div>
<div id="cdsbmexico-hands-on-help" class="section level4">
<h4>CDSBMexico hands on help</h4>
<p>If you need help directly, please get in touch with us via the CDSBMexico Slack workspace. If you either donâ€™t remember the Slack workspace or you couldnâ€™t attend CDSBMexico 2018, please get in touch with us via Twitter at <a href="https://twitter.com/CDSBMexico">twitter.com/CDSBMexico</a>.</p>
<p>We understand that not many will be able to attend BioC2019, so keep an eye open for CDSBMexico 2019! We are hoping to announce our plans soon.</p>
</div>
</div>
<div id="acknowledgments" class="section level3">
<h3>Acknowledgments</h3>
<p>This blog post was made possible thanks to:</p>
<ul>
<li><em><a href="https://bioconductor.org/packages/3.10/BiocStyle">BiocStyle</a></em> <a id='cite-Oles_2020'></a>(<a href='https://github.com/Bioconductor/BiocStyle'>OleÅ›, Morgan, and Huber, 2020</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=blogdown">blogdown</a></em> <a id='cite-Xie_2017'></a>(<a href='https://github.com/rstudio/blogdown'>Xie, Hill, and Thomas, 2017</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> <a id='cite-Boettiger_2019'></a>(<a href='https://CRAN.R-project.org/package=knitcitations'>Boettiger, 2019</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id='cite-Csardi_2018'></a>(<a href='https://CRAN.R-project.org/package=sessioninfo'>CsÃ¡rdi, core, Wickham, Chang, et al., 2018</a>)</li>
</ul>
</div>
<div id="references" class="section level3">
<h3>References</h3>
<p>
<a id='bib-Boettiger_2019'></a><a href="#cite-Boettiger_2019">[1]</a><cite>
C. Boettiger.
<em>knitcitations: Citations for â€˜Knitrâ€™ Markdown Files</em>.
R package version 1.0.10.
2019.
URL: <a href="https://CRAN.R-project.org/package=knitcitations">https://CRAN.R-project.org/package=knitcitations</a>.</cite>
</p>
<p>
<a id='bib-Csardi_2018'></a><a href="#cite-Csardi_2018">[2]</a><cite>
G. CsÃ¡rdi, R. core, H. Wickham, W. Chang, et al.
<em>sessioninfo: R Session Information</em>.
R package version 1.1.1.
2018.
URL: <a href="https://CRAN.R-project.org/package=sessioninfo">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id='bib-Oles_2020'></a><a href="#cite-Oles_2020">[3]</a><cite>
A. OleÅ›, M. Morgan, and W. Huber.
<em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>.
R package version 2.14.4.
2020.
URL: <a href="https://github.com/Bioconductor/BiocStyle">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id='bib-Xie_2017'></a><a href="#cite-Xie_2017">[4]</a><cite>
Y. Xie, A. P. Hill, and A. Thomas.
<em>blogdown: Creating Websites with R Markdown</em>.
ISBN 978-0815363729.
Boca Raton, Florida: Chapman and Hall/CRC, 2017.
URL: <a href="https://github.com/rstudio/blogdown">https://github.com/rstudio/blogdown</a>.</cite>
</p>
</div>
<div id="reproducibility" class="section level3">
<h3>Reproducibility</h3>
<pre><code>## â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
##  setting  value                       
##  version  R version 3.6.2 (2019-12-12)
##  os       macOS Catalina 10.15.2      
##  system   x86_64, darwin15.6.0        
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       America/New_York            
##  date     2020-02-12                  
## 
## â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
##  package       * version date       lib source                            
##  assertthat      0.2.1   2019-03-21 [1] CRAN (R 3.6.0)                    
##  bibtex          0.4.2.2 2020-01-02 [1] CRAN (R 3.6.0)                    
##  BiocManager     1.30.10 2019-11-16 [1] CRAN (R 3.6.1)                    
##  BiocStyle     * 2.14.4  2020-01-09 [1] Bioconductor                      
##  blogdown        0.17    2019-11-13 [1] CRAN (R 3.6.1)                    
##  bookdown        0.17    2020-01-11 [1] CRAN (R 3.6.0)                    
##  cli             2.0.1   2020-01-08 [1] CRAN (R 3.6.0)                    
##  colorout      * 1.2-1   2019-05-07 [1] Github (jalvesaq/colorout@7ea9440)
##  crayon          1.3.4   2017-09-16 [1] CRAN (R 3.6.0)                    
##  digest          0.6.23  2019-11-23 [1] CRAN (R 3.6.0)                    
##  evaluate        0.14    2019-05-28 [1] CRAN (R 3.6.0)                    
##  fansi           0.4.1   2020-01-08 [1] CRAN (R 3.6.0)                    
##  glue            1.3.1   2019-03-12 [1] CRAN (R 3.6.0)                    
##  htmltools       0.4.0   2019-10-04 [1] CRAN (R 3.6.0)                    
##  httr            1.4.1   2019-08-05 [1] CRAN (R 3.6.0)                    
##  jsonlite        1.6     2018-12-07 [1] CRAN (R 3.6.0)                    
##  knitcitations * 1.0.10  2019-09-15 [1] CRAN (R 3.6.0)                    
##  knitr           1.27    2020-01-16 [1] CRAN (R 3.6.0)                    
##  lubridate       1.7.4   2018-04-11 [1] CRAN (R 3.6.0)                    
##  magrittr        1.5     2014-11-22 [1] CRAN (R 3.6.0)                    
##  plyr            1.8.5   2019-12-10 [1] CRAN (R 3.6.0)                    
##  R6              2.4.1   2019-11-12 [1] CRAN (R 3.6.1)                    
##  Rcpp            1.0.3   2019-11-08 [1] CRAN (R 3.6.0)                    
##  RefManageR      1.2.12  2019-04-03 [1] CRAN (R 3.6.0)                    
##  rlang           0.4.3   2020-01-24 [1] CRAN (R 3.6.2)                    
##  rmarkdown       2.1     2020-01-20 [1] CRAN (R 3.6.0)                    
##  sessioninfo   * 1.1.1   2018-11-05 [1] CRAN (R 3.6.0)                    
##  stringi         1.4.5   2020-01-11 [1] CRAN (R 3.6.0)                    
##  stringr         1.4.0   2019-02-10 [1] CRAN (R 3.6.0)                    
##  withr           2.1.2   2018-03-15 [1] CRAN (R 3.6.0)                    
##  xfun            0.12    2020-01-13 [1] CRAN (R 3.6.0)                    
##  xml2            1.2.2   2019-08-09 [1] CRAN (R 3.6.0)                    
##  yaml            2.2.0   2018-07-25 [1] CRAN (R 3.6.0)                    
## 
## [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library</code></pre>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>Using <a href="https://www.google.com/flights">Google flights</a> for the search.<a href="#fnref1" class="footnote-back">â†©ï¸</a></p></li>
</ol>
</div>
